src/mpi/CubedSphereDistributedDiscreteModels.jl

```diff
@@ -0,0 +1,349 @@
+
+  Dc=2
+  PXEST_CORNERS=4
+  pXest_ghost = ptr_pXest_ghost[]
+  pXest = ptr_pXest[]
+
+  # Obtain ghost quadrants
+  ptr_ghost_quadrants = Ptr{P4est_wrapper.p4est_quadrant_t}(pXest_ghost.ghosts.array)
+
+  tree_offsets = unsafe_wrap(Array, pXest_ghost.tree_offsets, pXest_ghost.num_trees+1)
+  dcell_coordinates_and_panels=map_parts(parts) do part
+     ncells=pXest.local_num_quadrants+pXest_ghost.ghosts.elem_count
+     panels = Vector{Int}(undef,ncells)
+     data = Vector{Point{Dc,Float64}}(undef,ncells*PXEST_CORNERS)
+     ptr  = generate_ptr(ncells)
+     current=1
+     current_cell=1
+     vxy=Vector{Cdouble}(undef,Dc)
+     pvxy=pointer(vxy,1)
+     for itree=1:pXest_ghost.num_trees
+       tree = p4est_tree_array_index(pXest.trees, itree-1)[]
+       if tree.quadrants.elem_count > 0
+          set_panel_vertices_coordinates!( ptr_pXest_connectivity, coarse_discrete_model, itree)
+       end
+       for cell=1:tree.quadrants.elem_count
+          panels[current_cell]=itree
+          quadrant=p4est_quadrant_array_index(tree.quadrants, cell-1)[]
+          for vertex=1:PXEST_CORNERS
+            GridapP4est.p4est_get_quadrant_vertex_coordinates(ptr_pXest_connectivity,
+                                                  p4est_topidx_t(itree-1),
+                                                  quadrant.x,
+                                                  quadrant.y,
+                                                  quadrant.level,
+                                                  Cint(vertex-1),
+                                                  pvxy)
+            data[current]=Point{Dc,Float64}(vxy...)
+            current=current+1
+          end
+          current_cell=current_cell+1
+       end
+     end
+
+     # Go over ghost cells
+     for i=1:pXest_ghost.num_trees
+      if tree_offsets[i+1]-tree_offsets[i] > 0
+        set_panel_vertices_coordinates!( ptr_pXest_connectivity, coarse_discrete_model, i)
+      end
+      for j=tree_offsets[i]:tree_offsets[i+1]-1
+          panels[current_cell]=i
+          quadrant = ptr_ghost_quadrants[j+1]
+          for vertex=1:PXEST_CORNERS
+            GridapP4est.p4est_get_quadrant_vertex_coordinates(ptr_pXest_connectivity,
+                                                     p4est_topidx_t(i-1),
+                                                     quadrant.x,
+                                                     quadrant.y,
+                                                     quadrant.level,
+                                                     Cint(vertex-1),
+                                                     pvxy)
+
+          #  if (MPI.Comm_rank(comm.comm)==0)
+          #     println(vxy)
+          #  end
+          data[current]=Point{Dc,Float64}(vxy...)
+          current=current+1
+         end
+         current_cell=current_cell+1
+       end
+     end
+     Gridap.Arrays.Table(data,ptr), panels
+  end
+end
+
+function generate_cube_grid_geo(cell_coordinates_and_panels)
```

Comment:
Hi @amartinhuertas , Is this ordering of the panels consistently ordered? I think not right? (Perhaps I'm mistaken) We can consistently order the panels (so no sign_flip required) if we always travel in the same diagonal direction through the panels. Does that make sense?

Reply:
Hi @davelee2804, as far I remember, for this particular data structure (parallel cubed sphere), I reused the same ordering of panels that was used in the UMKO cubed sphere meshes files that you shared. As a result, in here  see https://github.com/gridapapps/GridapGeosciences.jl/pull/26/files I do not have to explicitly change the sign of the normal to get it pointing outwards for some of the panels, as I had to do with the sequential data structure in here https://github.com/gridapapps/GridapGeosciences.jl/blob/master/src/CubedSphereDiscreteModels.jl#L157

Thus, answering your question, I think that in the parallel data structure the panels are consistently oriented with respect to the sphere manifold, but not necessarily oriented equivalently as in the sequential data structure.

Reply:
Thanks for the clarification @amartinhuertas . If necessary we can always re-order, but great to know that it is already oriented.

Reply:
See also https://github.com/gridapapps/GridapGeosciences.jl/pull/26/files#diff-ade293314c9bbc1cbab7c1aa0f90c560811028373d99f66a8a9bc7fa6212f8f3R26 and attached picture 
![Screenshot from 2021-11-17 17-43-36](https://user-images.githubusercontent.com/38347633/142148134-0b1278d7-cd32-4190-a35d-34c1de9f935e.png)

Reply:
Thanks @amartinhuertas , I can see that if the two left panels are rotated pi/2 clockwise and the top panel is rotated pi/2 anti-clockwise the mesh is consistently oriented. Thanks for sharing.